### ISSUE TITLE
Goal: To work on Gaussian NLL on facial keypoint detection task

### ISSUE THREAD

**Issue by nandhini30** [RnDProjectsDeebul/NandhiniMathivananRnD#7]
Worked on it and have attached the loss graph in this comment.
![loss](https://user-images.githubusercontent.com/71918189/175358508-727ba74d-57d0-4da3-9dba-4a17382615f5.png)

**Comment by deebuls** [comment]
Great job !! Please continue the such kind of pro-active methods.

Feedback:
- Which loss are you tracking RMSE or NLL ?
- Please plot both for Gaussian as you need RMSE to compare with the non-uncertainty method
- Looking at the long tail one can see that you might be overlearning
    - Read about EarlyStopping and use that to stop the training early when there is no learning
- Make a [boxplot](https://seaborn.pydata.org/generated/seaborn.boxplot.html) and  [density plot](https://seaborn.pydata.org/generated/seaborn.kdeplot.html) for both the RMSE and the predicted sigma
- Calculate entropy of the predicted gaussian distribution and also make a [Density plot ](https://seaborn.pydata.org/generated/seaborn.kdeplot.html)
    - We need both this plot  
- Also take some sample images and plot the predicted points and the uncertainty as heatmap
    - This can be little tricky , just send me a set of images and the predicted output, I will do it fast.
    - Choose the 5 images with min error and 5 images with max error
    - Also choose 5 image with min sigma and 5 images with max sigma.

**Comment by deebuls** [comment]
@nandhini30  Dont wait to do all the task . Even if you have done any 1 task please put the corresponding graph here and we can discuss.

**Comment by nandhini30** [comment]
Okay sure. I’m working on earlystopping now. I’ll upload the graph once i finish that

**Comment by nandhini30** [comment]
I tried early stopping with patience = 3 for RMSE loss. This is the output i got
![loss](https://user-images.githubusercontent.com/71918189/178732095-887286df-d7b3-4b9c-8eec-481646c8df45.png)
Total no of epochs = 300 epochs, it stopped at epoch 138 epochs as early stopping.
I plotted val loss For RMSE
![RMSE_valloss_box](https://user-images.githubusercontent.com/71918189/178733171-52da1bf2-ec8e-4a12-ab24-d60385555f78.png)
![RMSE_valloss_dense](https://user-images.githubusercontent.com/71918189/178733182-fa363647-d847-4e90-87e2-f99460c2c165.png)

![RMSE_valloss_violin](https://user-images.githubusercontent.com/71918189/178733190-f6ea087f-7810-42e0-884e-14ae1b6fa611.png)
For gaussian loss, it will upload it before night

**Comment by nandhini30** [comment]
![GNLL_entropy_dense](https://user-images.githubusercontent.com/71918189/178821038-1596fc23-1280-44cd-ae56-39d1158bf17b.png)
![GNLL_valloss_box](https://user-images.githubusercontent.com/71918189/178821258-6a5c1ab6-3113-4540-90e8-5ba2e3ab8a84.png)
![GNLL_valloss_violin](https://user-images.githubusercontent.com/71918189/178821315-25e92e8d-9e86-48d6-b3c0-d2e4c0f86a97.png)

**Comment by deebuls** [comment]
@nandhini30  good work .

The only reason we plot is to compare(think about it !! There is no other purpose).
If you agree on the above , then the plots you have plotted what should I compare ? 
If the answer is nothing to compare then the plot is not serving the purpose.

Maybe plot with val_loss box plots of 2 methods . So that we can compare how 2 different methods loss. Etc. you can plot . 

Golden rule for plotting . Is there atleast 2 values to compare ? 

Also read at the above comments where I have given a list of comments .

**Comment by nandhini30** [comment]
Yes..i will send the compared plots today. I just wanted to update my progress. That’s why i uploaded the individual graphs

**Comment by nandhini30** [comment]
![concat_trainloss](https://user-images.githubusercontent.com/71918189/179763261-fb79f6b7-edbb-430b-a7e2-3d9f77ab104f.png)
![concat_trainloss](https://user-images.githubusercontent.com/71918189/179763278-bf6958cc-811b-4d09-8306-fc63f0e6b09a.png)
![concat_valloss](https://user-images.githubusercontent.com/71918189/179763279-0ca5d53c-9ae7-4323-a470-4e7bc3512628.png)
![concat_trainloss](https://user-images.githubusercontent.com/71918189/179763300-66aaa4e7-2d47-452a-a577-8c38abf9fe27.png)
![concat_valloss](https://user-images.githubusercontent.com/71918189/179763301-7eec1d88-64f4-4702-a74b-a743ae8c6a46.png)
![concat_violin_trainloss](https://user-images.githubusercontent.com/71918189/179763303-0dc43194-409b-465a-b4f8-8e40c285e2be.png)

**Comment by nandhini30** [comment]
I have attached the sprint report for 20 july, 2022 in this comment.
[Sprint report_20july.pdf](https://github.com/RnDProjectsDeebul/NandhiniMathivananRnD/files/9148236/Sprint.report_20july.pdf)

**Comment by nandhini30** [comment]
Below graph is drawn between |val_difference| vs two methods
![prediction](https://user-images.githubusercontent.com/71918189/180653769-c6c1f1f1-060f-4688-9d4b-f1d4cfa09ad5.png)
![prediction_violin](https://user-images.githubusercontent.com/71918189/180653772-1d66d877-de73-4c80-bdaf-10c578664023.png)
